test_requests.py

```diff
@@ -1177,6 +1181,24 @@ def test_precedence(self):
         encodings = requests.utils.get_encodings_from_content(content)
         assert encodings == ['HTML5', 'HTML4', 'XML']
 
+    def test_chunked_upload(self):
+        """ can safely send generators """
+        block_server = threading.Event()
+        server = Server.basic_response_server(wait_to_close_event=block_server)
+        data = (i for i in [b'a', b'b', b'c']) 
+
+        with server as (host, port):
+            url = 'http://{}:{}/'.format(host, port)
+            r = requests.post(url, data=data, stream=True)
+            block_server.set() # release server block
```

Comment:
We could have problems if the data upload takes more than 0.5 seconds (````consume_socket_content```` has a 0.5 timeout). I don't think that happens very often, but wanted the tests to be correct without assuming that the generator data is going to be sent without breaking any timeouts.